pandas/core/indexes/period.py

```diff
@@ -613,25 +611,21 @@ def get_loc(self, key, method=None, tolerance=None):
 
         try:
             key = Period(key, freq=self.freq)
+            ordinal = key.ordinal if key is not NaT else key.value
         except ValueError:
             # we cannot construct the Period
             # as we have an invalid type
             if is_list_like(key):
                 raise TypeError(f"'{key}' is an invalid key")
             raise KeyError(key)
 
-        ordinal = key.ordinal if key is not NaT else key.value
```

Comment:
i think this got affected by #31021.  unless there a compelling reason to move this line up to L614, lets keep it here